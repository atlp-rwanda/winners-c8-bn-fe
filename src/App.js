import React from 'react';
import Home from './components/Home';
import { Route, Routes } from 'react-router-dom';
import UserRegisterForm from './components/RegisterForm';
import RecoveryForm from './components/recoveryForm';
import ResetForm from './components/resetForm';
import LoginForm from './components/LoginForm';
import Userprofile from './components/UserProfile/UserProfile';
import Request from './pages/Request';
import Comments from './pages/Comments';
import Dashboard from './components/Dashboard';
import NotFound from './components/NotFound';
import UserRoleDash from "./components/AssignRole/UserRoleDash"
import { ToastContainer } from 'react-toastify';
import DashboardLayout from './components/Layouts/Dashboard';
import HomePageLayout from './components/Layouts/Home';

const App = () => {
  return (
    <div className="App">
      <ToastContainer />
      <div className="content">
        <Routes>
          <Route path="/" element={<HomePageLayout />}>
            <Route exact path="login" element={<LoginForm />} />
            <Route index element={<Home />} />
            <Route exact path="register" element={<UserRegisterForm />} />
            <Route exact path="recover" element={<RecoveryForm />} />
            <Route exact path="reset" element={<ResetForm />} />
          </Route>
          <Route path="/dashboard" element={<DashboardLayout />}>
            <Route exact index element={<Dashboard />} />
            <Route path="trips" element={<Request />} />
            <Route path="trips/trip" element={<Comments />} />
          <Route exact path='assignRole' element={<UserRoleDash />} />
            <Route exact path="userprofile" element={<Userprofile />} />
            <Route path="*" element={<NotFound />} />
          </Route>
          <Route exact path="*" element={<NotFound />} />
        </Routes>
      </div>
      {/* <SocialAuth /> */}
    </div>
  );
};

export default App;
