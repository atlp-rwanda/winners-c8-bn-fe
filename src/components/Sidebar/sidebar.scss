.sidebar {
    flex: 1;
    border-right: 0.5px solid rgb(230, 227, 227);
    min-height: 100vh;
    background-color: white;
  
    .top {
      height: 50px;
      display: flex;
      align-items: center;
      justify-content: center;
  
      .logo {
        position: absolute;
        top: 10px;
        left: 20px;
      }
    }
  
    hr {
      height: 0;
      border: 0.5px solid rgb(230, 227, 227);
    }
  
    .center {
      padding-left: 10px;

      ul {
        list-style: none;
        margin: 0;
        padding: 0;
  
        .title {
          font-size: 10px;
          font-weight: bold;
          color: #606060;
          margin-top: 35px;
          margin-bottom: 5px;
          font-style: normal;
          font-weight: 500;
          font-size: 11px;
          line-height: 16px;
        }
  
        li {
          display: flex;
          align-items: center;
          padding: 5px;
          cursor: pointer;
  
          &:hover {
            background-color: #ece8ff;
          }
  
          .icon {
            font-size: 18px;
            color:#395B64;
            ;
          }
  
          span{
              color: #606060;
              margin-left: 10px;
              font-family: 'Poppins';
              font-style: normal;
              font-weight: 400;
              font-size: 14px;
              line-height: 21px;
          }
        }
      }
    }
  
  }