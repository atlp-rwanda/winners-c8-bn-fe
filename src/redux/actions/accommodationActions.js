import axios from 'axios'
import { successToast, errorToast } from '../../helpers/generateToast';
import { successAlert, failureAlert} from './alertActions';
import { FETCH_ACCOMMODATIONS_LOADING, 
         FETCH_ACCOMMODATIONS_SUCCESS, 
         FETCH_ACCOMMODATIONS_FAILED,
         FETCH_SINGLE_ACCOMMODATION_LOADING,
         FETCH_SINGLE_ACCOMMODATION_SUCCESS,
         FETCH_SINGLE_ACCOMMODATION_FAILED} from "./actionTypes";
import {accommodationsUrl} from "../utils/apiUrls";
import {authHeader} from '../utils/dataSession';

export const listAccommodations = () => async dispatch => {
    dispatch({
        type: FETCH_ACCOMMODATIONS_LOADING
    });

  return await axios.get(accommodationsUrl, { headers: authHeader() })
      .then(res => {
          dispatch({
              type: FETCH_ACCOMMODATIONS_SUCCESS,
              payload: res.data
          });
      })
      .catch(err => {
          dispatch({
              type: FETCH_ACCOMMODATIONS_FAILED,
              payload: err.message,
              message: err
          });
          errorToast(err)
      })
}

export const detailsAccommodation = (accommodationId) => async dispatch => {
    dispatch({
        type: FETCH_SINGLE_ACCOMMODATION_LOADING
    });

  return await axios.get(accommodationsUrl+`${accommodationId}`, { headers: authHeader() })
      .then(res => {
          dispatch({
              type: FETCH_SINGLE_ACCOMMODATION_SUCCESS,
              payload: res.data
          });
      })
      .catch(err => {
          dispatch({
              type: FETCH_SINGLE_ACCOMMODATION_FAILED,
              payload: "failed to fetch",
              message: err
          });
      })
}
