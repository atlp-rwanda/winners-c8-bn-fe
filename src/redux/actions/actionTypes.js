export const FECTH_REQUESTS = 'FECTH_REQUESTS';
export const FECTH_REQUESTS_COMMENTS = 'FECTH_REQUESTS_COMMENTS';
export const FECTH_REQUESTS_COMMENTS_FAILED = 'FECTH_REQUESTS_COMMENTS_FAILED';
export const POST_REQUESTS_COMMENT = 'FECTH_REQUESTS_COMMENTS_FAILED';
export const POST_REQUESTS_COMMENT_FAILED = 'FECTH_REQUESTS_COMMENTS_FAILED';
export const DELETE_REQUESTS_COMMENT = 'DELETE_REQUESTS_COMMENT';
export const DELETE_REQUESTS_COMMENT_FAILED = 'DELETE_REQUESTS_COMMENT_FAILED';

export const GOOGLE_SIGN_IN_START = 'GOOGLE_SIGN_IN_START';
export const GOOGLE_SIGN_IN_SUCCESS = 'GOOGLE_SIGN_IN_SUCCESS';
export const GOOGLE_SIGN_IN_FAIL = 'GOOGLE_SIGN_IN_FAIL';

export const FACEBOOK_SIGN_IN_START = 'FACEBOOK_SIGN_IN_START';
export const FACEBOOK_SIGN_IN_SUCCESS = 'FACEBOOK_SIGN_IN_SUCCESS';
export const FACEBOOK_SIGN_IN_FAIL = 'FACEBOOK_SIGN_IN_FAIL';

export const FETCH_ACCOMMODATIONS_LOADING = 'FETCH_ACCOMMODATIONS_LOADING';
export const FETCH_ACCOMMODATIONS_SUCCESS = 'FETCH_ACCOMMODATIONS_SUCCESS';
export const FETCH_ACCOMMODATIONS_FAILED = 'FETCH_ACCOMMODATIONS_FAILED';

export const FETCH_SINGLE_ACCOMMODATION_LOADING =
  'FETCH_SINGLE_ACCOMMODATION_LOADING';
export const FETCH_SINGLE_ACCOMMODATION_SUCCESS =
  'FETCH_SINGLE_ACCOMMODATION_SUCCESS';
export const FETCH_SINGLE_ACCOMMODATION_FAILED =
  'FETCH_SINGLE_ACCOMMODATION_FAILED';

export const FETCH_USER_LOADING = 'FETCH_USER_LOADING';
export const FETCH_USER_SUCCESS = 'FETCH_USER_SUCCESS';
export const FETCH_USER_FAILED = 'FETCH_USER_FAILED';

export const FETCH_ROLE_LOADING = 'FETCH_ROLE_LOADING';
export const FETCH_ROLE_SUCCESS = 'FECTH_ROLE_SUCCESS';
export const FETCH_ROLE_FAILED = 'FECTH_ROLE_FAILED';

export const ASSIGN_ROLE_LOADING = 'ASSIGN_ROLE_LOADING';
export const ASSIGN_ROLE_SUCCESS = 'ASSIGN_ROLE_SUCCESS';
export const ASSIGN_ROLE_FAILED = 'ASSIGN_ROLE_FAILED';

export const LOGOUT_SUCCESS = 'LOGOUT_SUCCESS';
export const LOGOUT_FAILED = 'LOGOUT_FAILED';

export const ACCOMMODATION_UPDATE_LOADING = 'ACCOMMODATION_UPDATE_LOADING';
export const ACCOMMODATION_UPDATE_SUCCESS = 'ACCOMMODATION_UPDATE_SUCCESS';
export const ACCOMMODATION_UPDATE_FAILURE = 'ACCOMMODATION_UPDATE_FAILURE';

export const ACCOMMODATION_DELETE_LOADING = 'ACCOMMODATION_DELETE_LOADING';
export const ACCOMMODATION_DELETE_SUCCESS = 'ACCOMMODATION_DELETE_SUCCESS';
export const ACCOMMODATION_DELETE_FAILURE = 'ACCOMMODATION_DELETE_FAILURE';

export const ACCOMMODATION_VIEW_LOADING = 'ACCOMMODATION_VIEW_LOADING';
export const ACCOMMODATION_VIEW_SUCCESS = 'ACCOMMODATION_VIEW_SUCCESS';
export const ACCOMMODATION_VIEW_FAILURE = 'ACCOMMODATION_VIEW_FAILURE';
export const APPROVEREJECT_PENDING_REQUEST = 'APPROVE_PENDING_REQUEST';
export const APPROVEREJECT_SUCCESS = 'APPROVE_REQUEST_SUCCESS';
export const APPROVEREJECT_ERROR = 'APPROVE_REQUEST_ERROR';

export const FETCH_ALL_CHATS = 'GET_CHATS';
export const FETCH_ALL_CHATS_FAILED = 'FAILED_TO_LOAD_CHATS';
export const CHAT_SUCCESS = 'CHAT_SUCCESS';
export const CHATTING_STATUS = 'CHATTING_STATUS';
