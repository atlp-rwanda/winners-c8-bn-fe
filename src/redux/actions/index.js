// this folder will contain redux's actions