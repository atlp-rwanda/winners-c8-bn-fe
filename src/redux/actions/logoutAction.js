import {
   LOGOUT_FAILED,
   LOGOUT_SUCCESS
} from './actionTypes';
import { successToast, errorToast } from '../../helpers/generateToast';
import axiosInstance from '../../helpers/http';

export const logout = () => async (dispatch, navigate) => {
    return axiosInstance
      .put('/auth/signout')
      .then((res) => {
        if (res.data.error) {
           dispatch({
            type: LOGOUT_FAILED,
            payload: res.data.error,
          });
          errorToast(res.data.error);
        }
  
        if (res.status == 200) {
          if (window.localStorage.getItem('auth-token')) {
            window.localStorage.removeItem('auth-token');
          }
           dispatch({
            type: LOGOUT_SUCCESS,
            payload: res.data,
          });
          successToast(res.data.message);
            navigate('/login', {replace: true})


        }

  
      })
      .catch((err) => {
        dispatch({
          type: LOGOUT_FAILED,
          payload: err,
        });
        errorToast(err.response.data.error);
      });
  };