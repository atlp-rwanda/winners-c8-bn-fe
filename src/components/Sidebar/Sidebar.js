import React from "react";
import "./sidebar.scss";
import logo from "../../../public/images/logo.png";
import DashboardIcon from "@mui/icons-material/Dashboard";
import PersonOutlineIcon from "@mui/icons-material/PersonOutline";
import LocalShippingIcon from "@mui/icons-material/LocalShipping";
import CreditCardIcon from "@mui/icons-material/CreditCard";
import StoreIcon from "@mui/icons-material/Store";
import InsertChartIcon from "@mui/icons-material/InsertChart";
import SettingsApplicationsIcon from "@mui/icons-material/SettingsApplications";
import ExitToAppIcon from "@mui/icons-material/ExitToApp";
import NotificationsNoneIcon from "@mui/icons-material/NotificationsNone";
import SettingsSystemDaydreamOutlinedIcon from "@mui/icons-material/SettingsSystemDaydreamOutlined";
import PsychologyOutlinedIcon from "@mui/icons-material/PsychologyOutlined";
import AccountCircleOutlinedIcon from "@mui/icons-material/AccountCircleOutlined";
import { Link } from "react-router-dom";
import { useContext } from "react";

const Sidebar = () => {
  return (
    <div className="sidebar">
      <div className="top">
        <Link to="/" style={{ textDecoration: "none" }}>
          <img src={logo} alt="logo barefoot" className="logo" width={100} />
        </Link>
      </div>
      <hr />
      <div className="center">
        <ul>
          <p className="title">MAIN</p>
          <li>
            <DashboardIcon className="icon" />
            <span>Dashboard</span>
          </li>
          <p className="title">PAGES</p>
          <Link to="/users" style={{ textDecoration: "none" }}>
            <li>
              <PersonOutlineIcon className="icon" />
              <span>Create locat&accom</span>
            </li>
          </Link>
          <Link to="/products" style={{ textDecoration: "none" }}>
            <li>
              <StoreIcon className="icon" />
              <span>Assign Roles</span>
            </li>
          </Link>
          <li>
            <CreditCardIcon className="icon" />
            <span>Assign Managers</span>
          </li>
          <p className="title">COMPONENTS</p>
          <li>
            <InsertChartIcon className="icon" />
            <span>Tables</span>
          </li>
          <li>
            <NotificationsNoneIcon className="icon" />
            <span>Widgets</span>
          </li>
          <li>
            <NotificationsNoneIcon className="icon" />
            <span>Maps</span>
          </li>
          <p className="title">SOCIAL</p>
          <li>
            <SettingsSystemDaydreamOutlinedIcon className="icon" />
            <span>Notifications</span>
          </li>
          <li>
            <PsychologyOutlinedIcon className="icon" />
            <span>Chatting</span>
          </li>
          <li>
            <SettingsApplicationsIcon className="icon" />
            <span>Settings</span>
          </li>
          <p className="title">GENERAL</p>
          <li>
            <AccountCircleOutlinedIcon className="icon" />
            <span>Profile</span>
          </li>
          <li>
            <ExitToAppIcon className="icon" />
            <span>Logout</span>
          </li>
        </ul>
      </div>
      <div className="bottom">
        {/* color dark mode/ */}
      </div>
    </div>
  );
};

export default Sidebar;