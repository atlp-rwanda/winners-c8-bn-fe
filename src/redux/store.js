// this file will configure the redux's store
import { createStore } from "redux";
import rootReducer from "./reducers";

let authToken = ""; 
if(window.localStorage.getItem("auth-token")) {
  authToken = window.localStorage.getItem("auth-token");
}

const initialState = {
    auth: {token: authToken},
    navbar: {currentPage: "HOME"},
};
const store = createStore(
    rootReducer,
    initialState // sets initial state
  );

export default store;
