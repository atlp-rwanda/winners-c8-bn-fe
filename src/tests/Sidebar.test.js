import React from "react";
import Sidebar from "../components/Sidebar/Sidebar";
import renderer from 'react-test-renderer'
import { logout } from "../redux/actions/logoutAction";
import { cleanup,render,screen, act, fireEvent } from "@testing-library/react";
import '@testing-library/jest-dom';
import store from '../redux/store';
import App from '../App';
import { BrowserRouter as Router } from 'react-router-dom';
import { MemoryRouter} from 'react-router-dom';
import { Provider } from "react-redux";
import thunk from "redux-thunk";
import MockAxiosAdapter from 'axios-mock-adapter';
import axios from "axios";
const mock = new MockAxiosAdapter(axios)



describe('Sidebar Dashboard' , () =>{
    beforeEach(() =>{
     mock.onPut(/logout/gi).reply(200);
    })
    afterEach(() => {
        cleanup();
        window.localStorage.clear();
        mock.reset();
      }); 

      it('dispatches SUCCESS after logout', async () => {
        const userObj = {
          email: 'honore@gmail.com',
          password: 'Password@12345',
        };
    
        await store.dispatch(logout(userObj));
      });
      test('should render Sidebar Component', () => {
        render(
          <Provider store={store}>
            <Router>
              <Sidebar />
            </Router>
          </Provider>
        );
  const SidebarElement = screen.getByTestId('sidebar-test');
  expect(SidebarElement).toBeInTheDocument()
  expect(SidebarElement).toHaveTextContent('Dashboard');
      });

      it('The page should render  the page and logout', async () => {
        window.localStorage.setItem('auth-token', 'testing');
        await act(() =>
        render(
          <Provider store={store}>
            <MemoryRouter initialEntries={['/auth/logout']}>
              <App />
            </MemoryRouter>
          </Provider>
        )
      );
    
      })
      test("<Sidebar /> matches snapshot", ()=>{   
        const component = renderer
              .create(
                  <Provider store={store}>
                      <MemoryRouter>
                <Sidebar />
                      </MemoryRouter>
                  </Provider>,
              )
              .toJSON();
          expect(component).toMatchSnapshot();
      })
})
