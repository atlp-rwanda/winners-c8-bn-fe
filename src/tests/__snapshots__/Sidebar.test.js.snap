// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar Dashboard <Sidebar /> matches snapshot 1`] = `
<div
  className="sidebar"
  data-testid="sidebar-test"
>
  <div
    className="top"
  >
    <a
      href="/"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <img
        alt="logo barefoot"
        className="logo"
        src=""
        width={100}
      />
    </a>
  </div>
  <hr />
  <div
    className="center"
  >
    <ul>
      <p
        className="title"
      >
        MAIN
      </p>
      <a
        href="/dashboard"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <li>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
            data-testid="DashboardIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 13h8V3H3v10zm0 8h8v-6H3v6zm10 0h8V11h-8v10zm0-18v6h8V3h-8z"
            />
          </svg>
          <span>
            Dashboard
          </span>
        </li>
      </a>
      <p
        className="title"
      >
        PAGES
      </p>
      <a
        href="/dashboard/accommodations"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <li>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
            data-testid="PersonOutlineIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 5.9c1.16 0 2.1.94 2.1 2.1s-.94 2.1-2.1 2.1S9.9 9.16 9.9 8s.94-2.1 2.1-2.1m0 9c2.97 0 6.1 1.46 6.1 2.1v1.1H5.9V17c0-.64 3.13-2.1 6.1-2.1M12 4C9.79 4 8 5.79 8 8s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 9c-2.67 0-8 1.34-8 4v3h16v-3c0-2.66-5.33-4-8-4z"
            />
          </svg>
          <span>
            Create locat&accom
          </span>
        </li>
      </a>
      <a
        href="/dashboard/assignRole"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <li>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
            data-testid="StoreIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 4H4v2h16V4zm1 10v-2l-1-5H4l-1 5v2h1v6h10v-6h4v6h2v-6h1zm-9 4H6v-4h6v4z"
            />
          </svg>
          <span>
            Assign Roles
          </span>
        </li>
      </a>
      <li>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
          data-testid="CreditCardIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 4H4c-1.11 0-1.99.89-1.99 2L2 18c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V6c0-1.11-.89-2-2-2zm0 14H4v-6h16v6zm0-10H4V6h16v2z"
          />
        </svg>
        <span>
          Assign Managers
        </span>
      </li>
      <p
        className="title"
      />
      <a
        href="/dashboard/trips"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <li>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
            data-testid="InsertChartIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z"
            />
          </svg>
          <span>
            Trips
          </span>
        </li>
      </a>
      <li>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
          data-testid="NotificationsNoneIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.9 2 2 2zm6-6v-5c0-3.07-1.63-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.64 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2zm-2 1H8v-6c0-2.48 1.51-4.5 4-4.5s4 2.02 4 4.5v6z"
          />
        </svg>
        <span>
          Widgets
        </span>
      </li>
      <li>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
          data-testid="NotificationsNoneIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.9 2 2 2zm6-6v-5c0-3.07-1.63-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.64 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2zm-2 1H8v-6c0-2.48 1.51-4.5 4-4.5s4 2.02 4 4.5v6z"
          />
        </svg>
        <span>
          Maps
        </span>
      </li>
      <p
        className="title"
      >
        SOCIAL
      </p>
      <li>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
          data-testid="SettingsSystemDaydreamOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 17H9c-2.21 0-4-1.79-4-4 0-1.93 1.36-3.56 3.22-3.92C9.04 7.8 10.47 7 12 7c1.95 0 3.66 1.28 4.26 3.09 1.58.36 2.74 1.75 2.74 3.41 0 1.93-1.57 3.5-3.5 3.5zm-6.76-5.98C7.74 11.15 7 11.99 7 13c0 1.1.9 2 2 2h6.5c.83 0 1.5-.67 1.5-1.5s-.67-1.5-1.5-1.5h-.87l-.17-.86C14.29 9.92 13.23 9 12 9c-.96 0-1.84.57-2.26 1.45l-.27.57h-.73zM21 3H3c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h18c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 16.01H3V4.99h18v14.02z"
          />
        </svg>
        <span>
          Notifications
        </span>
      </li>
      <li>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
          data-testid="PsychologyOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m15.82 7.22-1 .4c-.21-.16-.43-.29-.67-.39L14 6.17c-.02-.1-.1-.17-.2-.17h-1.6c-.1 0-.18.07-.19.17l-.15 1.06c-.24.1-.47.23-.67.39l-1-.4c-.09-.03-.2 0-.24.09l-.8 1.38c-.05.09-.03.2.05.26l.85.66c-.03.12-.05.26-.05.39s.01.26.03.39l-.84.66c-.08.06-.1.17-.05.25l.8 1.39c.05.09.15.12.25.09l.99-.4c.21.16.43.29.68.39l.14 1.06c.02.1.1.17.2.17h1.6c.1 0 .18-.07.2-.17l.15-1.06c.24-.1.47-.23.67-.39l.99.4c.09.04.2 0 .24-.09l.8-1.39c.05-.09.03-.19-.05-.25l-.83-.66c.02-.13.03-.26.03-.39 0-.14-.01-.27-.03-.39l.85-.66c.08-.06.1-.17.05-.26l-.8-1.38c-.05-.09-.16-.12-.25-.09zM13 11.43c-.79 0-1.43-.64-1.43-1.43s.64-1.43 1.43-1.43 1.43.64 1.43 1.43-.64 1.43-1.43 1.43z"
          />
          <path
            d="M19.94 9.06c-.43-3.27-3.23-5.86-6.53-6.05C13.27 3 13.14 3 13 3 9.47 3 6.57 5.61 6.08 9l-1.93 3.48c-.41.66.07 1.52.85 1.52h1v2c0 1.1.9 2 2 2h1v3h7v-4.68c2.62-1.25 4.35-4.08 3.94-7.26zm-5.05 5.57-.89.42V19h-3v-3H8v-4H6.7l1.33-2.33C8.21 7.06 10.35 5 13 5c2.76 0 5 2.24 5 5 0 2.09-1.29 3.88-3.11 4.63z"
          />
        </svg>
        <span>
          Chatting
        </span>
      </li>
      <li>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
          data-testid="SettingsApplicationsIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm7-7H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-1.75 9c0 .23-.02.46-.05.68l1.48 1.16c.13.11.17.3.08.45l-1.4 2.42c-.09.15-.27.21-.43.15l-1.74-.7c-.36.28-.76.51-1.18.69l-.26 1.85c-.03.17-.18.3-.35.3h-2.8c-.17 0-.32-.13-.35-.29l-.26-1.85c-.43-.18-.82-.41-1.18-.69l-1.74.7c-.16.06-.34 0-.43-.15l-1.4-2.42c-.09-.15-.05-.34.08-.45l1.48-1.16c-.03-.23-.05-.46-.05-.69 0-.23.02-.46.05-.68l-1.48-1.16c-.13-.11-.17-.3-.08-.45l1.4-2.42c.09-.15.27-.21.43-.15l1.74.7c.36-.28.76-.51 1.18-.69l.26-1.85c.03-.17.18-.3.35-.3h2.8c.17 0 .32.13.35.29l.26 1.85c.43.18.82.41 1.18.69l1.74-.7c.16-.06.34 0 .43.15l1.4 2.42c.09.15.05.34-.08.45l-1.48 1.16c.03.23.05.46.05.69z"
          />
        </svg>
        <span>
          Settings
        </span>
      </li>
      <p
        className="title"
      >
        GENERAL
      </p>
      <a
        href="/dashboard/userprofile"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <li>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
            data-testid="AccountCircleOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM7.35 18.5C8.66 17.56 10.26 17 12 17s3.34.56 4.65 1.5c-1.31.94-2.91 1.5-4.65 1.5s-3.34-.56-4.65-1.5zm10.79-1.38C16.45 15.8 14.32 15 12 15s-4.45.8-6.14 2.12C4.7 15.73 4 13.95 4 12c0-4.42 3.58-8 8-8s8 3.58 8 8c0 1.95-.7 3.73-1.86 5.12z"
            />
            <path
              d="M12 6c-1.93 0-3.5 1.57-3.5 3.5S10.07 13 12 13s3.5-1.57 3.5-3.5S13.93 6 12 6zm0 5c-.83 0-1.5-.67-1.5-1.5S11.17 8 12 8s1.5.67 1.5 1.5S12.83 11 12 11z"
            />
          </svg>
          <span>
            Profile
          </span>
        </li>
      </a>
      <li>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
          data-testid="ExitToAppIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10.09 15.59 11.5 17l5-5-5-5-1.41 1.41L12.67 11H3v2h9.67l-2.58 2.59zM19 3H5c-1.11 0-2 .9-2 2v4h2V5h14v14H5v-4H3v4c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
        <span
          onClick={[Function]}
        >
          Logout
        </span>
      </li>
    </ul>
  </div>
</div>
`;
